src/imitation/scripts/parallel.py

```diff
@@ -115,16 +138,26 @@ def inner(config: dict, reporter) -> dict:
         }
         ex = experiments[sacred_ex_name]
 
-        observer = FileStorageObserver.create("sacred")
+        observer = FileStorageObserver("sacred")
         ex.observers.append(observer)
 
-        # Apply base configs
-        base_named_configs.extend(config.get("named_configs", []))
-        base_config_updates.update(config.get("config_updates", {}))
-        config["named_configs"] = base_named_configs
-        config["config_updates"] = base_config_updates
+        # Apply base configs to get modified `named_configs` and `config_updates`.
+        named_configs = []
+        named_configs.extend(base_named_configs)
+        named_configs.extend(run_kwargs["named_configs"])
+        updated_run_kwargs["named_configs"] = named_configs
+
+        config_updates = {}
+        config_updates.update(base_config_updates)
+        config_updates.update(run_kwargs["config_updates"])
+        updated_run_kwargs["config_updates"] = config_updates
+
+        # Add other run_kwargs items to updated_run_kwargs.
+        for k, v in run_kwargs.items():
```

Comment:
To check my understanding: the semantics here are that `base_config_updates` should take priority over `run_kwargs` when both specify the same key?

Reply:
No, it's actually the opposite. Because `base_config_updates` gets `update`-applied first, and then `run_kwargs["named_configs"]` overwrites this with the second `update`.

Looking back at the docstring, this doesn't seem well explained.
- [ ] Better docs for base_*_config

Reply:
Added better docs.

Reply:
This for loop adds all keys except "config_updates" and "named_configs" from the search_space into the run_kwargs. Hopefully the docs should clarify this a bit.

It's pretty messy getting the base_config_updates + config_updates in through the ray.tune API.